test/csi/toolkit/counter_tool.cpp

```diff
@@ -1,28 +1,27 @@
+#include "csi.h"
 #include <stdio.h>
 #include <stdlib.h>
 
+extern "C" {
 long times_accessed_by_size[4];
 
-static void __csi_destroy() {
+WEAK void __csi_destroy() {
     fprintf(stderr, "bytes accessed: %ld\n", times_accessed_by_size[0]);
     fprintf(stderr, "shorts accessed: %ld\n", times_accessed_by_size[1]);
     fprintf(stderr, "ints accessed: %ld\n", times_accessed_by_size[2]);
     fprintf(stderr, "longs accessed: %ld\n", times_accessed_by_size[3]);
 }
-extern "C" void __csi_init() {
+
+WEAK void __csi_init() {
```

Comment:
Do they have to be `WEAK` here? It was my impression (though I could be wrong) that the null tool could mark them as `WEAK`, and we define them here in the counter tool as strong symbols (to avoid any weirdness of, e.g., linking two tools by accident)